src/DynamoCoreWpf/UI/GuidedTour/CustomRichTextBox.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Documents;
+using System.Windows.Media.Imaging;
+
+namespace Dynamo.Wpf.UI.GuidedTour
+{
+    /// <summary>
+    /// This class represent a RichTextBox that will be used in Popups to show hyperlinks, texts, images, bullet points, gifs and other media items.
+    /// </summary>
+    public class CustomRichTextBox : RichTextBox
+    {
+
+        #region CustomText Dependency Property
+
+        public static readonly DependencyProperty CustomTextProperty = DependencyProperty.Register("CustomText", typeof(string), typeof(CustomRichTextBox),
+       new PropertyMetadata(string.Empty, CustomTextChangedCallback), CustomTextValidateCallback);
+
+        private static void CustomTextChangedCallback(
+            DependencyObject obj, DependencyPropertyChangedEventArgs e)
+        {
+            (obj as CustomRichTextBox).Document = GetCustomDocument(e.NewValue as string);
+        }
+
+        private static bool CustomTextValidateCallback(object value)
+        {
+            return value != null;
+        }
+
+        /// <summary>
+        /// This Custom property is used for registing the dependency property
+        /// </summary>
+        public string CustomText
+        {
+            get
+            {
+                return (string)GetValue(CustomTextProperty);
+            }
+            set
+            {
+                SetValue(CustomTextProperty, value);
+            }
+        }
+
+        #endregion
+
+
+        public CustomRichTextBox()
+        {
+
```

Comment:
Would you put some comments? e.g. // Do nothing for now. 

Empty function body will result in SonarQube scan issue

Reply:
@QilongTang  the constructor was removed since is not used, by default a public constructor will be created automatically.
Thanks